src/timer/windows/SDL_systimer.c

```diff
@@ -146,19 +146,31 @@ SDL_GetPerformanceFrequency(void)
 void
 SDL_Delay(Uint32 ms)
 {
-    /* Sleep() is not publicly available to apps in early versions of WinRT.
-     *
-     * Visual C++ 2013 Update 4 re-introduced Sleep() for Windows 8.1 and
-     * Windows Phone 8.1.
-     *
-     * Use the compiler version to determine availability.
-     *
-     * NOTE #1: _MSC_FULL_VER == 180030723 for Visual C++ 2013 Update 3.
-     * NOTE #2: Visual C++ 2013, when compiling for Windows 8.0 and
-     *    Windows Phone 8.0, uses the Visual C++ 2012 compiler to build
-     *    apps and libraries.
-     */
-#if defined(__WINRT__) && defined(_MSC_FULL_VER) && (_MSC_FULL_VER <= 180030723)
+    /* CREATE_WAITABLE_TIMER_HIGH_RESOLUTION flag was added in Windows 10 version 1803.
+    * 
+    * Sleep() is not publicly available to apps in early versions of WinRT.
+    *
+    * Visual C++ 2013 Update 4 re-introduced Sleep() for Windows 8.1 and
+    * Windows Phone 8.1.
+    *
+    * Use the compiler version to determine availability.
+    *
+    * NOTE #1: _MSC_FULL_VER == 180030723 for Visual C++ 2013 Update 3.
+    * NOTE #2: Visual C++ 2013, when compiling for Windows 8.0 and
+    *    Windows Phone 8.0, uses the Visual C++ 2012 compiler to build
+    *    apps and libraries.
+    */
+#if NTDDI_VERSION >= NTDDI_WIN10_RS4
+    HANDLE timer = CreateWaitableTimerExW(NULL, NULL, CREATE_WAITABLE_TIMER_HIGH_RESOLUTION, TIMER_ALL_ACCESS);
+    if (timer)
+    {
+        LARGE_INTEGER due_time;
+        due_time.QuadPart = -(LONGLONG)(ms * 10000);
+        if (SetWaitableTimerEx(timer, &due_time, 0, NULL, NULL, NULL, 0))
+            WaitForSingleObject(timer, INFINITE);
+        CloseHandle(timer);
+}
```

Comment:
What is the overhead of creating a new timer object every time you want to wait?

Reply:
I don't know exactly, but it is probably very light because every single code example where I saw timers being used to sleep were doing it like this. That being said, it's still probably better to keep the timer open and reuse it instead of recreating it. I'll try it and if everything works fine, I'll push it to this PR.

Reply:
Would reusing work considering multiple threads can call SDL_Delay concurrently?

You can probably use the SDL_GetPerformanceCounter and SDL_GetPerformanceFrequency APIs to measure the cost of creation.

Reply:
Ah you're right, we need to consider multithreading. According to the documentation,  ["Multiple processes can have handles to the same timer object, enabling use of the object for interprocess synchronization"](https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-createwaitabletimerexw?f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(SYNCHAPI%252FCreateWaitableTimerExW)%3Bk(CreateWaitableTimerExW)%3Bk(DevLang-C%252B%252B)%3Bk(TargetOS-Windows)%26rd%3Dtrue) but resetting the timer with a new delay will impact all the thread waiting on it : ["If the timer is already active when you call SetWaitableTimerEx, the timer is stopped, then it is reactivated. Stopping the timer in this manner does not set the timer state to signaled, so threads blocked in a wait operation on the timer remain blocked. However, it does cancel any pending completion routines."](https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-setwaitabletimerex). So if we want to reuse the timers, we'll indeed have to create one per thread.

That said, it may be entirely unnecessary to do it. I benchmarked the cumulated cost of timers creation and destruction with SDL_GetPerformanceCounter as you suggested, and the measured overhead was 1.8 µs on average. Considering that the accuracy of the timers is far higher (500 µs in the very best case), I would call this overhead negligible.